[LBT-42] add Syslog backend for loot-log

pasqu4le opened the pull request:
Adds a Syslog backend implementation for loot-log.

Includes an example configuration: `docs/example/syslog.yaml`

The `loot-log` README may need to be modified.

flyingleafe commented on code/log/lib/Loot/Log/Syslog.hs:
```diff
@@ -0,0 +1,299 @@
+        "PID"        -> pure SL.PID
+        "PERROR"     -> pure SL.PERROR
+        "withID"     -> pure SL.PID
+        "withStderr" -> pure SL.PERROR
```
Not sure if providing aliases for the same options is actually a good idea. A person who sets up the syslog is likely to be familiar with standard option flag names (`PID` and `PERROR`) anyway, and those aliases are not much more descriptive.

pasqu4le commented:
@flyingleafe replying here because Github is not building the pages yet:

> You could avoid this boilerplate by using makeLensesWith TH function.

Sure thing, I did't to avoid importing `microlens-th`, will fix.

> Not sure if providing aliases for the same options is actually a good idea. A person who sets up the syslog is likely to be familiar with standard option flag names (PID and PERROR) anyway, and those aliases are not much more descriptive.

I was actually thinking about this myself this very morning and I was going to get rid of `outOptions` altogether  in favor of both `withPID` and `withPERROR` fields. This would avoid the repetition of re-specifying the first (that is `True` by default) in yaml files when one wants to activate the second (that is `False` by default) and it will also make useless the re-export of `System.Log.Handler.Syslog`'s `Option`. What do you think?

pasqu4le commented:
Update: went with simple `makeLenses` and underscore-starting fields and replaced `outOptions` as I was explaining. Also squashed it all. Waiting for approval.
